experiments/llm-eval/main.py

```diff
@@ -31,45 +31,54 @@ def setup_mcp_config(config_data, config_file_path="~/.gemini/settings.json"):
         json.dump(config_data, f, indent=4)
     print(f"MCP configuration written to: {expanded_path}")
 
-def discover_tasks(tasks_dir="tasks"):
+def discover_tasks(tasks_dir="tasks", specific_task=None):
     """
-    Discovers test cases from subdirectories in the tasks directory.
+    Discovers test cases from immediate subdirectories in the tasks directory.
     """
     test_cases = []
-    for root, dirs, files in os.walk(tasks_dir):
-        if "task.yaml" in files:
-            task_path = os.path.join(root, "task.yaml")
-            with open(task_path, 'r') as f:
-                try:
-                    task_data = yaml.safe_load(f)
-                    test_name = os.path.basename(root)
-                    
-                    # Ensure script is a list
-                    script = task_data.get("script", [])
-                    if not isinstance(script, list):
-                        script = [script]
-
-                    # The prompt is the first item in the script list
-                    prompt = script[0].get("prompt") if script and "prompt" in script[0] else None
-
-                    if not prompt:
-                        print(f"Warning: No prompt found for task {test_name}, skipping.")
-                        continue
-                    
-                    output_file = task_data.get("output_file", f"{test_name}.yaml")
-                    prompt += f"\nPlace the generated YAML in a file at `$(git rev-parse --show-toplevel)/.build/{root}/{output_file}`."
-                    prompt += "\nHint: Use the available MCP tools to find the resource schema and generate the YAML. If the mcp tool is not available, use your knowledge to generate the YAML manifest."
-
-                    test_cases.append({
-                        "name": test_name,
-                        "prompt": prompt,
-                        "verifier_script": task_data.get("verifier"),
-                        "cleanup_script": task_data.get("cleanup"),
-                        "setup_script": "setup.sh" if "setup.sh" in files else None,
-                        "task_dir": root,
-                    })
-                except yaml.YAMLError as e:
-                    print(f"Error parsing YAML in {task_path}: {e}")
+    try:
+        with os.scandir(tasks_dir) as entries:
+            for entry in entries:
+                if not entry.is_dir():
+                    continue
+
+                task_dir = entry.path
+                task_name = entry.name
+
+                if specific_task and task_name != specific_task:
```

Comment:
My 2c: easier to load everything and filter afterwards.  But just my opinion, no need to fix